src/library/commissioner_impl.cpp

```diff
@@ -1955,6 +1962,314 @@ ByteArray CommissionerImpl::GetCommissionerDatasetTlvs(uint16_t aDatasetFlags)
     return tlvTypes;
 }
 
+ByteArray CommissionerImpl::GetNetDiagTlvTypes(uint64_t aDiagTlvFlags)
+{
+    ByteArray tlvTypes;
+
+    if (aDiagTlvFlags & NetDiagData::kExtMacAddressBit)
+    {
+        EncodeTlvType(tlvTypes, tlv::Type::kNetworkDiagExtMacAddress);
+    }
+
+    if (aDiagTlvFlags & NetDiagData::kMacAddressBit)
+    {
+        EncodeTlvType(tlvTypes, tlv::Type::kNetworkDiagMacAddress);
+    }
+
+    if (aDiagTlvFlags & NetDiagData::kModeBit)
+    {
+        EncodeTlvType(tlvTypes, tlv::Type::kNetworkDiagMode);
+    }
+
+    if (aDiagTlvFlags & NetDiagData::kRoute64Bit)
+    {
+        EncodeTlvType(tlvTypes, tlv::Type::kNetworkDiagRoute64);
+    }
+
+    if (aDiagTlvFlags & NetDiagData::kLeaderDataBit)
+    {
+        EncodeTlvType(tlvTypes, tlv::Type::kNetworkDiagLeaderData);
+    }
+
+    if (aDiagTlvFlags & NetDiagData::kIpv6AddressBit)
+    {
+        EncodeTlvType(tlvTypes, tlv::Type::kNetworkDiagIpv6Address);
+    }
+
+    if (aDiagTlvFlags & NetDiagData::kMacCountersBit)
+    {
+        EncodeTlvType(tlvTypes, tlv::Type::kNetworkDiagMacCounters);
+    }
+
+    if (aDiagTlvFlags & NetDiagData::kChildTableBit)
+    {
+        EncodeTlvType(tlvTypes, tlv::Type::kNetworkDiagChildTable);
+    }
+
+    if (aDiagTlvFlags & NetDiagData::kEui64Bit)
+    {
+        EncodeTlvType(tlvTypes, tlv::Type::kNetworkDiagEui64);
+    }
+
+    if (aDiagTlvFlags & NetDiagData::kChildIpv6AddressBit)
+    {
+        EncodeTlvType(tlvTypes, tlv::Type::kNetworkDiagChildIpv6Address);
+    }
+
+    return tlvTypes;
+}
+
+Error CommissionerImpl::DecodeNetDiagData(NetDiagData &aNetDiagData, const ByteArray &aPayload)
+{
+    Error       error;
+    tlv::TlvSet tlvSet;
+    NetDiagData diagData;
+    // Clear all data fields
+    diagData.mPresentFlags = 0;
```

Comment:
nit: this can be removed given the default value for `mPresentFlags` is already `0`